### PR TITLE
support: drop trailing slash for real

### PR DESCRIPTION
**Pull request by tylershunt**
`+` and `?` are both greedy, but the `+` operator is apparently more
greedy. To make this regex actually drop this slash when it's around
I need another `?` which is how you tell `+` not to be greedy.

Signed-off-by: Tyler Hunt <thunt@katanagraph.com>

### CODE REVIEW

File: libsupport/src/Uri.cpp [hunk 1]
```diff
@@ -17,7 +17,7 @@
 namespace {
 
 // get scheme and path, always drop trailing slash
-const std::regex kUriRegex("(?:([a-zA-Z0-9]+)://)?(.+)/?");
+const std::regex kUriRegex("^(?:([a-zA-Z0-9]+)://)?(.+?)/?$");
```
Comment (ddn0) [inline libsupport/src/Uri.cpp:20]: Can you add a unit test?

**Review by ddn0 — APPROVED** [review]

File: libsupport/src/Uri.cpp (hunk 1 above)
Comment (tylershunt) [reply libsupport/src/Uri.cpp:20]: Done.

File: libsupport/src/Uri.cpp [hunk 2]
```diff
@@ -17,7 +17,7 @@
...
```
Comment (witchel) [inline libsupport/src/Uri.cpp:21]: Why did you add the start (`^`) and end (`$`) markers?
